test/tests/functional/pbs_apbasil_inventory.py

```diff
@@ -0,0 +1,223 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2017 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+from string import Template
+import os
+import tempfile
+import xml.etree.ElementTree as ET
+
+
+class TestBasilInventory(TestFunctional):
+    """
+    This test suite is for testing the support for BASIL 1.4 inventory
+    query.Test if query is made with correct BASIL version, and that
+    vnodes are getting created as per the query response.
+    """
+    basil_version = ['1.4', '1.3', '1.2', '1.1']
+
+    def restartPBS(self):
+        try:
+            svcs = PBSInitServices()
+            svcs.restart()
+        except PbsInitServicesError, e:
+            self.logger.error("PBS restart failed: \n" + e.msg)
+            self.assertTrue(e.rv)
+
+    def reset_nodes(self, hostA):
+        # Remove all nodes
+        rv = self.server.manager(MGR_CMD_DELETE, NODE, None, "")
+        self.assertEqual(rv, 0)
+
+        # Restart PBS
+        self.restartPBS()
+
+        # Create node
+        rv = self.server.manager(MGR_CMD_CREATE, NODE, None, hostA)
+        self.assertEqual(rv, 0)
+
+        # Wait for 3 seconds for changes to take effect
+        time.sleep(3)
+
+    def setUp(self):
+        momA = self.mom
+        mom_config = momA.parse_config()
+        if '$alps_client' not in mom_config:
```

Comment:
why don't you use mom.is_cray? it has same code

Reply:
is_cray function checks for "alps_client" mom log, but the above check is 
for confirming the alps_client parameter is set in mom_priv/config.

Reply:
I'd still suggest you to use that because that is the supported way to detect Cray platform. If you think that is_cray is implemented wrongly then we should log that as a problem and get PTL fixed

Reply:
@dilip-krishnan I am Cray expert but AFAIK if you have "alps_client" in mom config then mom prints log message regarding that (something like 'set alps_client to ...') in its log file so not directly but indirectly mom.is_cray is checking alps_client in mom config...
But as @arungrover suggested if you think mom.is_cray is not doing right thing then please log a issue for that.

Reply:
@hirenvadalia ,
    The check i have put is not to see whether platform is cray or not, but to see if $alps_client is set in mom config.

Reply:
@dilip-krishnan But we set $alps_client in cray platform only right? and if you have set $alps_client in mom config then mom will print log message about that right?
Also if you are not checking for cray platform means your testcase will work on non-cray platforms also?